services/src/test/java/io/scalecube/services/sut/GreetingServiceImpl.java

```diff
@@ -12,7 +13,8 @@
 
 public final class GreetingServiceImpl implements GreetingService {
 
-  @Inject Microservices ms;
+  @Inject
+  MicroservicesContext ms;
```

Comment:
Can we revert it? Use Microservices instead of its context.

Reply:
Why?

Reply:
I like to have the `Microservices` instance. Will do it still work if I request `Microservices`? And if no then why?

Reply:
@artem-v help, it is your idea :) What is there in microservices that microservicesContext does not? The idea was to limit Microservices to an interface that only provides what you really need. I think the question is more philosophical.

Reply:
i kind of agree with @segabriel  the idea of having microservices is to grant access to the service to lower levels apis of scalecube.

Reply:
this access gives exactly MicroservicesContext

Reply:
@ronenhamias @segabriel I have now split MicroservicesContext into MicroservicesContext and ExtendedMicroservicesContext because we cannot access ServiceDiscovery during the ServiceDefinition collection phase. 

I'm sure you think this is a controversial step, but it's done specifically to prevent NPEs when using ServiceDiscovery in custom ServiceFactory implementations.

Reply:
https://github.com/scalecube/scalecube-services/pull/711#discussion_r437977735

Yes
The purpose of `MicroservicesContext` is to hide `Microservices` from client code. So service developer can see only some view of functions/objects from Microservices, plus it will depend only on service-api module, which is somewhat cleaner than to pull entire services library.

Reply:
@ronenhamias @segabriel btw question about _ExtendedMicroservicesContext_ is not relevant anymore, it will be one `MicroservicesContext`.